Updated documentation - Visual Studio specificities, Shield compat table, what's new update

IoTDan opened the pull request:
New documentation for a shield compatibility table and info on some "new to visual studio" specificities.

IoTDan committed:
Updated documentation for VS and shields

Updated What's new, added a Shield compat matrix, added some info about
Visual Studio Specificities.

IoTDan committed:
Updated Documentation after rendering review

spelling, broken links, table format

IoTDan committed:
doc updates - format fixes

IoTDan committed:
table fixes

IoTDan committed:
table spacing fixes

IoTDan committed:
Added release note about USB host

ooeygui commented on ShieldCompat.md:
```diff
@@ -0,0 +1,40 @@
+| [Universal 16-key Keypad](http://www.amazon.com/Universial-Switch-Keypad-Keyboard-Arduino/dp/B008A30NW4/ref=cm_cr_pr_product_top){:target="_blank"} | [Arduino Keypad Library](http://playground.arduino.cc/code/Keypad){:target="_blank"} | [Key Matricies Info](http://pcbheaven.com/wikipages/How_Key_Matrices_Works/){:target="_blank"} |
+
+<br/>
+<a class="btn btn-default" href="index.htm" role="button">Return to homepage</a>
```
## Can you update so it uses the markdown version of this?

`[&laquo; Return to Samples](SampleApps.htm){: .btn .btn-default}`

IoTDan replied:
updated.
[&laquo; Return to Homepage](index.htm){: .btn .btn-default}

ooeygui commented on VisualStudio.md:
```diff
@@ -0,0 +1,27 @@
+You can find more info about these files [here](http://msdn.microsoft.com/en-us/library/h552b3ca.aspx){:target="_blank"}.
+
+<br/>
+<a class="btn btn-default" href="index.htm" role="button">Return to homepage</a>
```
`[&laquo; Return to Samples](SampleApps.htm){: .btn .btn-default}`

IoTDan replied:
updated! 
[&laquo; Return to Homepage](index.htm){: .btn .btn-default}

ooeygui commented on VisualStudio.md:
```diff
@@ -0,0 +1,27 @@
+They make compilation even more efficient than using #pragmas.
+
+The default Wiring template creates a few files (stdafx.h and stdafx.cpp) in your project that are used to manage precompiled headers.
+You can find more info about these files [here](http://msdn.microsoft.com/en-us/library/h552b3ca.aspx){:target="_blank"}.
```
We've explicitly disabled them, so you should mention that. We can't enable them due to Arduino headers not including them

IoTDan replied:
Done!!!!

IoTDan committed:
Code reviews changes

1) Used Markdown version of button
2) Added that precompiled headers are disabled by default

IoTDan committed:
fixed spacing

IoTDan committed:
updated newlines
